### PR TITLE
478 family income batch edit

### LINKED ISSUE #478: family income batch edit form
- refactor family income form stuff to be like enrollment/funding 
  - reusable craete/edit form lives in `Components/Forms`
  - reference this form in flow-specific  (editRecord, batchEdit) wrappers etc when necessary

### PR DESCRIPTION
**Pull request by jlhogan**
closes #478

### CODE REVIEW

File: client/src/components/Forms/FamilyIncome/Form.tsx [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+          incomeDeterminationFields
+        ) && (
+          <div>
+            <DeterminationDateField />
+          </div>
+        )}
+      </FormFieldSet>
+
+      {CancelButton}
+      <FormSubmitButton text={loading ? 'Saving... ' : 'Save'} />
```
Comment (mmazanec22) [inline client/src/components/Forms/FamilyIncome/Form.tsx:142]: Should we offload this button to the component library to make sure save/saving text is consistent everywhere in the app?

File: client/src/components/Forms/types.ts [hunk 2]
```diff
@@ -14,5 +14,5 @@ export type RecordFormProps = {
     fields: string[],
     allFormFields: string[]
   ) => boolean;
-  AdditionalButton?: JSX.Element;
+  AdditionalButton?: JSX.Element; // Optional 'Cancel' (for forms in cards or card expansions) or 'Skip' (for forms in BatchEdit steplist) button
```
Comment (mmazanec22) [inline client/src/components/Forms/types.ts:17]: Yay comment!

File: client/src/containers/CreateRecord/listSteps.tsx [hunk 3]
```diff
@@ -18,7 +18,7 @@ export const newForms = [
   { key: SECTION_KEYS.IDENT, form: ChildIdentifiersForm },
```
Comment (mmazanec22) [inline client/src/containers/CreateRecord/listSteps.tsx:18]: If these and the edit record/whatever else keys and forms have converged completely, we can hopefully simplify down to one set of section key/form pairs

File: client/src/containers/EditRecord/FamilyIncome/EditDeterminationCard.tsx [hunk 4]
```diff
@@ -101,7 +107,18 @@ export const IncomeDeterminationCard = ({
...
+      <CardExpansion>
+        <FamilyIncomeForm
+          child={child}
+          id={`edit-determination-${determinationId}`}
+          legend={`Edit income determiniation ${determinationId}`}
+          afterSaveSuccess={() => {
+            setCloseCard(true);
+            afterSaveSuccess();
+          }}
+          setAlerts={() => {}}
```
Comment (mmazanec22) [inline client/src/containers/EditRecord/FamilyIncome/EditDeterminationCard.tsx:119]: Should we ditch this setAlerts or make it optional on the child maybe?

**Review by mmazanec22 — APPROVED** [review]

File: client/src/containers/EditRecord/FamilyIncome/EditDeterminationCard.tsx (hunk 4 above)
Comment (jlhogan) [reply client/src/containers/EditRecord/FamilyIncome/EditDeterminationCard.tsx:119]: that or we should use the alerts consistently to display error / success state in all form worlds

File: client/src/containers/CreateRecord/listSteps.tsx (hunk 3 above)
Comment (jlhogan) [reply client/src/containers/CreateRecord/listSteps.tsx:18]: they look the same but are not the same b/c the forms are imported from different places ... `EnrollmentFundingForm` is context-specific but at least now is a wrapped around the same internal form chunks (mostly)

File: client/src/components/Forms/FamilyIncome/Form.tsx (hunk 1 above)
Comment (jlhogan) [reply client/src/components/Forms/FamilyIncome/Form.tsx:142]: I believeeeeee the button already is in the compnent library with default text "save" -- but I think we should probably update the button to take in a `saving` boolean, so it does this consistently, yes !

**Review by bamader — APPROVED** [review]
LGTM!

File: client/src/components/Forms/FamilyIncome/Form.tsx (hunk 1 above)
Comment (mmazanec22) [reply client/src/components/Forms/FamilyIncome/Form.tsx:142]: Seeing changes to this PR and realizing that I meant to say "...to the component folder in this project" not the component library-- smaller lift, less commitment :)

**Pull request merged**